From c9d8e646ed505a9cd4c84b2cb31d846b8588109f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sat, 13 May 2023 18:37:57 +0200
Subject: [PATCH] fix missing <cstdint> include (#6720)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix missing <cstdint> include in src/util/tptr.h that causes build
failure with GCC 13:

```
In file included from /tmp/z3/src/util/region.cpp:53:
/tmp/z3/src/util/region.cpp: In member function ‘void* region::allocate(size_t)’:
/tmp/z3/src/util/tptr.h:29:62: error: ‘uintptr_t’ does not name a type
   29 | #define ALIGN(T, PTR) reinterpret_cast<T>(((reinterpret_cast<uintptr_t>(PTR) >> PTR_ALIGNMENT) + \
      |                                                              ^~~~~~~~~
/tmp/z3/src/util/region.cpp:82:22: note: in expansion of macro ‘ALIGN’
   82 |         m_curr_ptr = ALIGN(char *, new_curr_ptr);
      |                      ^~~~~
/tmp/z3/src/util/region.cpp:57:1: note: ‘uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   56 | #include "util/page.h"
  +++ |+#include <cstdint>
   57 |
```
---
 src/util/tptr.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util/tptr.h b/src/util/tptr.h
index 2a35af535..50e9417fe 100644
--- a/src/util/tptr.h
+++ b/src/util/tptr.h
@@ -19,6 +19,7 @@ Revision History:
 
 #pragma once
 
+#include <cstdint>
 #include "util/machine.h"
 #include <cstdint>
 
-- 
2.41.0

